polars/polars-lazy/polars-plan/src/dsl/arity.rs

```diff
@@ -106,17 +114,12 @@ impl WhenThenThen {
                 falsy: Box::new(otherwise),
             }
         }
-        if then_iter.next().is_some() {
```

Comment:
Shouldn't we check this anymore?

Reply:
Because `WhenThenThen` is split into two structs, it's now impossible to end up with an invalid chain. You can only call `when` after a `then` call and vice versa, and you can only call `otherwise` after a `then` call. So this check is no longer needed.